samcli/commands/init/interactive_init_flow.py

```diff
@@ -323,6 +325,49 @@ def _generate_from_use_case(
         )
 
 
+def _get_latest_python_runtime() -> str:
+    """
+    Returns the latest support version of Python
+    SAM CLI supports
+
+    Returns
+    -------
+    str:
+        The name of the latest Python runtime (ex. "python3.12")
+    """
+
+    # set python3.9 as fallback
+    latest_major = 3
+    latest_minor = 9
+
+    compiled_regex = re.compile(r"python(.*?)\.(.*)")
+
+    for runtime in SUPPORTED_RUNTIMES:
+        if not runtime.startswith("python"):
+            continue
+
+        # python3.12 => 3.12 => (3, 12)
+        version_match = re.match(compiled_regex, runtime)
+
+        if not version_match:
+            LOG.debug(f"Failed to match version while checking {runtime}")
+            continue
+
+        matched_groups = version_match.groups()
+
+        try:
+            version_major = int(matched_groups[0])
+            version_minor = int(matched_groups[1])
+        except (ValueError, IndexError):
+            LOG.debug(f"Failed to parse version while checking {runtime}")
+            continue
+
+        latest_major = version_major if version_major > latest_major else latest_major
+        latest_minor = version_minor if version_minor > latest_minor else latest_minor
+
+    return f"python{latest_major}.{latest_minor}"
```

Comment:
Should we log the version before we return?

Reply:
We can, I can log it as a debug message since info would immediately log this in the middle of the interactive flow. The customer already sees the latest determined version too (made a change below to now include this in the click prompt).